client/lib/service-worker/service-worker.js

```diff
@@ -100,3 +115,124 @@ self.addEventListener( 'message', function( event ) {
 			break;
 	}
 } );
+
+self.addEventListener( 'fetch', function( event ) {
+	const request = event.request;
+
+	if ( request.method !== 'GET' ) {
+		return;
+	}
+
+	// HTML Pages, fetch from the server and fallback to the Login page
```

Comment:
this may not be a good strategy longer term. We're working on building log-in off of a different entrypoint than the rest of the app.

That said, the normal root doesn't have anything fancy in it, except maybe the bootstrapped current user... What's the problem with using it?